tests/default_test.go

```diff
@@ -0,0 +1,61 @@
+package test
+
+import (
+	_ "WeatherAPI/routers"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/astaxie/beego"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func init() {
+	_, file, _, _ := runtime.Caller(1)
+	log.Print(file)
+	// apppath, _ := filepath.Abs(filepath.Dir(filepath.Join(file, ".."+string(filepath.Separator))))
+	apppath := "/mnt/c/Users/carlosalberto.rojas/Desktop/go/src/WeatherAPI/"
+	// apppath := "\\mnt\\c\\Users\\carlosalberto.rojas\\Desktop\\go\\src\\WeatherAPI\\"
+
+	log.Print("apppath: ")
+	log.Print(apppath)
+	log.Print(string(filepath.Separator))
+
+	beego.TestBeegoInit(apppath)
+}
+
+// TestGet is a sample to run an endpoint test
+func TestGet(t *testing.T) {
+	r, _ := http.NewRequest("GET", "/v1/object", nil)
+	w := httptest.NewRecorder()
+	beego.BeeApp.Handlers.ServeHTTP(w, r)
+
+	beego.Trace("testing", "TestGet", "Code[%d]\n%s", w.Code, w.Body.String())
+
+	Convey("Subject: Test Station Endpoint\n", t, func() {
+		Convey("Status Code Should Be 200", func() {
+			So(w.Code, ShouldEqual, 200)
+		})
+		Convey("The Result Should Not Be Empty", func() {
+			So(w.Body.Len(), ShouldBeGreaterThan, 0)
+		})
+	})
+}
+
+func TestWeather(t *testing.T) {
+
+	Convey("Given a HTTP request for /weather Without GET params", t, func() {

```

Comment:
@carojaspy please take a look of table driven tests and please add more test cases for all the errors you got in the controller. 

Hint. Simulate failures from the external service.

Reply:
Yes I am, same as your comment about hitting the external service every time, I worked it on v3,  I could make a "cherry pick" to take that test to this branch.